### DISCUSSION TITLE
Block Classes not triggered

### DISCUSSION THREAD

**Discussion by michaeljcoyne** [area17/twill#1623] [General]
Hi 

Looking at the docs

https://twill.io/docs/block-editor/rendering-blocks.html#modifying-block-data

I have a block class file in correct namespace and a block blade file.

Set up as instructed.

Ran twill:blocks and build.
cleared cache

tried to print out getData(); and then die in my block class.

It is not triggered.

Is there a step I am missing?

**Comment by haringsrob** [comment]
Hey @michaeljcoyne ,

You should not need to print getData, basically everything you return from the getData method will be available as a variable.

**Comment by michaeljcoyne** [comment]
@haringsrob  I am just trying to get this working. The  print_r($data); die; Just means it is not loading for me.

**Reply by haringsrob** [reply]
Ah like that I understand, What is the class/namespace and the name of the view?

Also, are you on the latest twill version?

**Comment by michaeljcoyne** [comment]
Twill version 2.8.0

namespace App\Twill\Block;

Class path: app/Twill/Block/RepeaterAgenda.php

Block path: resources/views/admin/blocks/repeater_agenda.blade.php
Block path: resources/views/site/blocks/repeater_agenda.blade.php

```namespace App\Twill\Block;

use A17\Twill\Services\Blocks\Block;

class RepeaterAgenda extends Block
{
    public function getData(array $data, \A17\Twill\Models\Block $block): array
    {
        $data = parent::getData($data, $block);
        print_r($data); die;

        return $data;
    }
}
```

**Reply by haringsrob** [reply]
I think the path should be: Class path: app/Twill/**Block**/RepeaterAgenda.php

**Reply by michaeljcoyne** [reply]
@haringsrob sorry typo - corrected - and version no.

**Reply by haringsrob** [reply]
Strange, it should work. I will test this out tomorrow with these filenames to see what's going on.

**Reply by michaeljcoyne** [reply]
Seems like it is not autoloading

**Reply by haringsrob** [reply]
you can try composer dump-autoload, but in general that should not be needed.

**Comment by haringsrob** [comment]
Hey @michaeljcoyne,

You did everything according to the docs, which is perfect. However, I noticed that the most recent docs are not yet live which is why it is not working.

Everything is fine but you do need to suffix the name of the block class with `Block` so the full class is:

`app/Twill/Block/RepeaterAgendaBlock.php` after that everything should work.

I will make sure that docs are updated.
